core/src/main/java/hudson/Plugin.java

```diff
@@ -79,6 +82,22 @@
 public abstract class Plugin implements Saveable {
 
     /**
+     * You do not need to create custom subtypes:
+     * <ul>
+     * <li>{@code config.jelly}, {@link #configure(StaplerRequest, JSONObject)}, {@link #load}, and {@link #save}
+     *      can be replaced by {@link GlobalConfiguration}
+     * <li>{@link #start} and {@link #postInitialize} can be replaced by {@link Initializer} (or {@link ItemListener#onLoaded})
+     * <li>{@link #stop} can be replaced by {@link Terminator}
+     * <li>{@link #setServletContext} can be replaced by {@link Jenkins#servletContext}
+     * </ul>
+     * Note that every plugin gets a {@link DummyImpl} by default,
+     * which will still route the URL space, serve {@link #getWrapper}, and so on.
+     * @deprecated Use more modern APIs rather than subclassing.
+     */
+    @Deprecated
+    protected Plugin() {}
```

Comment:
:bug:  May cause regressions in the existing plugin code, which utilizes public constructors. Not sure the case is realistic.

I would vote for adding a public constructor and marking it as Restricted